src/Streamly/Internal/Data/Stream/StreamK.hs

```diff
@@ -784,6 +792,22 @@ zipWithM f = go
             yield1 = merge
         foldStream (adaptState st) yield1 single1 stp mx
 
+-- | Zip two streams serially using a monadic zipping function. Run the streams
+-- in parallel.
+--
+-- /Unreleased/
+{-# INLINE zipAsyncWithM #-}
+zipAsyncWithM :: MonadAsync m =>
+    (a -> b -> m c) -> Stream m a -> Stream m b -> Stream m c
+zipAsyncWithM f m1 m2 = mkStream $ \st yld sng stp -> do
+    sv1 <- newParallelVar StopNone (adaptState st)
+    sv2 <- newParallelVar StopNone (adaptState st)
+    SVar.toSVarParallel (adaptState st) sv1 $ D.fromStreamK m1
+    SVar.toSVarParallel (adaptState st) sv2 $ D.fromStreamK m2
+    let pm1 = getSerialT $ SVar.fromSVar sv1
+        pm2 = getSerialT $ SVar.fromSVar sv2
+    foldStream st yld sng stp $ zipWithM f pm1 pm2
```

Comment:
Can we create an SVar only for the second stream? So that we evaluate the first stream in the current thread and the second stream in a child thread. That way we do not have to create an additional thread.